### PR TITLE
Port AWS libraries to use options/lifecycle.

### PR DESCRIPTION
**Pull request by jason-bragg**
This updates the AWS libraries use options and lifecycle rather than the legacy provider model.
Work was done by @galvesribeiro, I'm just shepherding it through the process.

### CODE REVIEW

**Comment by ReubenBond** [comment]
Please undo the changes which add underscores to field names

File: test/AWSUtils.Tests/Streaming/SQSStreamTests.cs [hunk 1]
```diff
@@ -48,14 +48,15 @@ protected override void ConfigureTestCluster(TestClusterBuilder builder)
...
                 };
-                options.ClusterConfiguration.Globals.RegisterStorageProvider<DynamoDBStorageProvider>("DynamoDBStore", storageConnectionString);
-                var storageConnectionString2 = new Dictionary<string, string>
-                {
-                    {"DataConnectionString", $"Service={AWSTestConstants.Service}"},
-                    {"DeleteStateOnClear", "true"},
-                    {"UseJsonFormat", "true"}
-                };
-                options.ClusterConfiguration.Globals.RegisterStorageProvider<DynamoDBStorageProvider>("PubSubStore", storageConnectionString2);
+                // FIXME: How to configure TestClusterBuilder with the extension methods
```
Comment (sergeybykov) [inline test/AWSUtils.Tests/Streaming/SQSStreamTests.cs]: Why is this commented out? What's here to fix?

Comment (galvesribeiro) [reply test/AWSUtils.Tests/Streaming/SQSStreamTests.cs]: I didn't found how to properly use the extension methods to register the provider using the `TestCluster`.

Comment (jason-bragg) [reply test/AWSUtils.Tests/Streaming/SQSStreamTests.cs]: Updated this to setup the providers correctly (I think).

@galvesribeiro, can vet that is setup the secrets correctly?  There isn't a data connection string in the new options..?

File: test/AWSUtils.Tests/Streaming/SQSStreamTests.cs [hunk 2]
```diff
@@ -70,7 +62,17 @@ public void Configure(ISiloHostBuilder hostBuilder)
                     .AddSqsStreams("SQSProvider2", options =>
                      {
                          options.ConnectionString = AWSTestConstants.DefaultSQSConnectionString;
-                     });
+                     })
+                    .AddDynamoDBGrainStorage("DynamoDBStore", options =>
+                    {
+                        options.SecretKey = $"Service={AWSTestConstants.Service}";
```
Comment (jason-bragg) [inline test/AWSUtils.Tests/Streaming/SQSStreamTests.cs]: @galvesribeiro, am I setting these options correctly?  I this where they need be set, I just don't know which options to set..

Comment (galvesribeiro) [reply test/AWSUtils.Tests/Streaming/SQSStreamTests.cs]: Sorry, mislooked that part. Just need to set `options.Service` to the `AWSTestConstants.Service`. That property has the emulator URL already there. https://github.com/jason-bragg/orleans/blob/c1a19f8ced75735db95ecd1ecf8d856ade42fc3a/test/AWSUtils.Tests/StorageTests/AWSTestConstants.cs#L45-L47

**Comment by jason-bragg** [comment]
fixed test failures, though they looked unrelated.  Not sure how this change would have caused them..?

File: src/AWS/Orleans.Clustering.DynamoDB/Options/DynamoDBClusteringOptions.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        /// <summary>
+        /// AccessKey string for DynamoDB Storage
+        /// </summary>
+        public string AccessKey { get; set; }
+
+        /// <summary>
+        /// Secret key for DynamoDB storage
+        /// </summary>
+        public string SecretKey { get; set; }
```
Comment (ReubenBond) [inline src/AWS/Orleans.Clustering.DynamoDB/Options/DynamoDBClusteringOptions.cs]: Add `[RedactConnectionString]`

Comment (ReubenBond) [reply src/AWS/Orleans.Clustering.DynamoDB/Options/DynamoDBClusteringOptions.cs]: Here and above

Comment (jason-bragg) [reply src/AWS/Orleans.Clustering.DynamoDB/Options/DynamoDBClusteringOptions.cs]: agreed, not sure how I missed those.  Waiting for tests to complete before pushing change, to see if here are any other tests I need to fix.

**Comment by jason-bragg** [comment]
Tests pass.  Pushing a fix to options formatting, should be able to merge after that.

**Comment by jason-bragg** [comment]
Updated.  Ready for commit.

**Pull request merged**